repair/repair.cc

```diff
@@ -2143,7 +2140,9 @@ future<> repair_service::repair_tablets(repair_uniq_id rid, sstring keyspace_nam
             shard_id master_shard_id;
             // Repair all tablets belong to this node
             for (auto& r : replicas) {
-                if (select_primary_ranges_within_dc) {
+                // Check primary replica only within local dc
+                // since there's no global primary replica with tablets
+                if (primary_replica_only) {
```

Comment:
Benny, this is incorrect.  This is going to break nodetool repair -pr and cause over repair. I can give you an simple example. 

Say, we have n1,n2,n3 in dc1, n4,n5,n6 in dc2. RF = 2 in each DC. 

Given a tablet, t1 with n1, n2, n4, n6. 

When we run nodetool repair -pr on all nodes. 

With your change,

When n1 runs nodetool repair -pr, it checks n1 is the primary replica. n1 will start a repair to repair t1 for replicas on n1,n2,n4,n6.

When n4 runs nodetool repair -pr, it checks n4 is the primary replica (It skips dc1). n4  will start a repair to repair t1 for replicas on n1,n2,n4,n6.

This mean the tablet t1 will be repaired twice. It defeats the purpose of running with -pr completely.